src/main/scala/temple/builder/DatabaseBuilder.scala

```diff
@@ -0,0 +1,57 @@
+package temple.builder
+
+import temple.DSL.semantics.{Annotation, Attribute, AttributeType, ServiceBlock}
+import temple.generate.database.ast.ColumnConstraint.Check
+import temple.generate.database.ast._
+
+/** Construct database queries from a Templefile structure */
+object DatabaseBuilder {
+
+  private def generateMaxMinConstraints[T](name: String, max: Option[T], min: Option[T]): Seq[ColumnConstraint] = {
+    val maxCondition = max.map(v => Check(name, ComparisonOperator.LessEqual, v.toString))
+    val minCondition = min.map(v => Check(name, ComparisonOperator.GreaterEqual, v.toString))
+    Seq(maxCondition, minCondition).flatten
+  }
+
+  private def toColDef(name: String, attribute: Attribute): ColumnDef = {
+    val valueConstraints = attribute.valueAnnotations.map {
+      case Annotation.Unique => ColumnConstraint.Unique
+    }.toSeq
+
+    val (colType, constraints) = attribute.attributeType match {
+      case AttributeType.BoolType     => (ColType.BoolCol, valueConstraints)
```

Comment:
Personally I feel that using `Nil` here as the `typeConstraint`, and adding on `valueConstraints` afterwards, would be less repetitive than including `valueConstraints` on every line